common/http/src/metrics_endpoint.rs

```diff
@@ -0,0 +1,39 @@
+use futures::TryFutureExt;
+use hyper::{
+    header::CONTENT_TYPE,
+    service::{make_service_fn, service_fn},
+    Body, Request, Response, Server,
+};
+use prometheus::{Encoder, TextEncoder};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error(transparent)]
+    Server(#[from] hyper::Error),
+}
+
+pub async fn serve(port: &u32) -> Result<(), Error> {
```

Comment:
extremely minor, but ports are usually u16s. It's not worth changing though.

Reply:
good point... got me thinking about the actual setting exposed.
I'll change it to `u16` everywhere.